text/0000-scaling-use-cases.md

```diff
@@ -0,0 +1,115 @@
+
+Another solution might be to artificially generate transactions every three users.
+
+<!-- ##### -->
+<!-- ##### -->
+
+## Global Wallet with Derivation Paths
+
+This case is similar to the Global Wallet. But, instead of associating one index per user, it associates one derivation path per user. For instance:
+
+```
+derivation_path | user
+----------------|----------
+  m/44'/0'/0'/0 | Alice
+  m/44'/0'/0'/1 | Bob
+  m/44'/0'/0'/2 | Charlie
+  m/44'/0'/0'/3 | Dave
+  m/44'/0'/0'/4 | Eve
+```
+
+Notice that each user has one, and only one, derivation path.
+
+Each derivation path leads to an unlimited number of enumerable addresses. Thus, each user has addresses #0, #1, #2, #3, and so on. Notice that Alice's address #0 is different from Bob's address #0 because they had different derivation paths.
+
+This approach has a similar scaling issue of the Global Wallet. The wallet will have to keep track of all these derivation paths and their addresses.
+
+Another challenge is to scan the derivation paths. The most common stop criteria for the scanner is the GAP Limit M/N, which stops the scanner when `M` derivation paths in a row are empty. It considers that a given derivation path is empty when its first `N` addresses are empty.
+
+## One Wallet Per User
+
+This case differs from the previous ones because there will be multiple wallets running.
+
+First of all, there are multiple ways to associate a wallet with a user. To illustrate, some ways are: (i) one seed per user, (ii) one global seed with one passphrase per user, or (iii) one global seed with one derivation path per user.
+
+The challenges here are the same regardless the way the wallet is associated with a user.
+
+The naive implementation here is to run one wallet per user. But it would consume a lot of resources (CPU and memory) and it is hard to scale to millions of users.
+
+A better implementation is to run a pool of wallets that automatically starts and stops wallets on demand. For instance, when a request is made to a wallet that is not running, the pool starts that wallet automatically. When a wallet is idle for too long, it is automatically stopped.
+
+Such pool would consume resources according to the number of concurrent users (instead of total users). It can easily scale up through the addition of more servers to the pool.
+
+## Full-length Wallet Service
+
+A Full-Length Wallet Service is a service that keeps track of all addresses in the blockchain and allows the creation of wallets.
+
+Wallets can have two types: manual or xpub.
+
+- Manual wallets have a list of addresses that is manually maintened, i.e., one must manually add or remove addresses from the wallet.
+- Xpub wallets have an xpub, and the addresses are scanned using the GAP Limit stop criteria.
+
+In this case, it is easier to use xpub wallets and ensure that the deposits meets the requirements of the GAP Limit.
+
```

Comment:
We could add a section explaining a bit of how HD Wallets work (BIP32 and BIP44)